demisto_sdk/commands/run_cmd/runner.py

```diff
@@ -312,3 +339,97 @@ def execute_command(self, command: str):
         context = ast.literal_eval(context)
 
         return res, context
+
+    def get_server_type(self) -> ServerType:
+        if not self.client.api_client.configuration.host.startswith("https://api-"):
+            return ServerType.XSOAR
+        raw_response, _, response_headers = self.client.generic_request(
+            "/about", "GET", response_type="object"
+        )
+        if "text/html" in response_headers.get("Content-Type"):
+            raise ValueError(
+                "",  # TODO
+            )
+        logger.debug(f"about={raw_response}")
+
+        if (
+            raw_response.get("productMode", "") == "xsoar"
+            and raw_response.get("deploymentMode", "") == "saas"
+        ):
+            return ServerType.XSOAR_SAAS
+        return ServerType.XSIAM
+
+    def create_incident_for_query(self):
+        """
+        Creates an incident if the server is XSOAR SaaS and returns the incident ID.
+        Otherwise, no incident is created, and None is returned.
+        """
+
+        # Check if the server is XSOAR 8
+        if self.server_type is not ServerType.XSOAR_SAAS:
+            self.is_incident_runtime_created = False
+            return
+
+        incident, status_code, _ = self.client.generic_request(
+            path="/incident",
+            method="POST",
+            content_type="application/json",
+            body={
+                "createInvestigation": True,
+                "name": "Demisto-sdk Run",
+                "owner": "",
+                "severity": 0,
+                "type": "Unclassified",
+                "playbook": "",
+            },
+            response_type=object,
+        )
+
+        if status_code != 201:
+            logger.critical(
+                "Failed to create a new incident for executing the command."
+            )
```

Comment:
Can we add the raw response from the `client.generic_request` in the log in case of a non-201 response? It could help with debugging in the future.